src/compositions/cascade/useCascadePanels.ts

```diff
@@ -0,0 +1,178 @@
+import { computed, ComputedRef, inject, InjectionKey, MaybeRefOrGetter, provide, reactive, readonly, Ref, ref, toRef } from 'vue'
+import { isDefined } from '@/utilities'
+
+export const cascadePanelsKey: InjectionKey<UseCascadePanels> = Symbol('UseCascadePanels')
+
+export type CascadePanelId = string | symbol | number
+export type CascadeState = Record<CascadePanelId, boolean>
+
+export type CascadePanel = {
+  id: CascadePanelId,
+  level?: number,
+}
+
+export function getInjectedCascadePanels(): UseCascadePanels {
+  const cascadePanels = inject(cascadePanelsKey)
+
+  if (!cascadePanels) {
+    throw new Error('Cascade panels not found. Are you sure the component calling useCascadePanels() exists within a <p-cascade> component context?')
+  }
+
+  return cascadePanels
+}
```

Comment:
IMO its a little bit of a red flag that this is an exported utility but it must be run in a component context. I'd personally rename this to be a composition so that there's no confusion.

Reply:
Hm that's fair, it's only really meant for internal use (of these components) since `useCascade*` compositions return this already. That said, I agree that's a little confusing so I renamed; I could remove the export altogether in favor of calling the injection directly in the `useCascadePanel` composition (that's why this is exported in the first place) - I'd want to duplicate the error handling though, which isn't a huge deal 🤷🏻

Reply:
If it was me I would duplicate the logic but create an error class and throw that error. Not exporting the utility has more pros than the loss of dryness.

Reply:
Works for me then - updated 👍🏻